### ISSUE TITLE
`emits_changed_signal` for `dbus_interface` properties

### ISSUE THREAD

**Issue by zeenix** [dbus2/zbus#323]
While !479 added support of `org.freedesktop.DBus.Property.EmitsChangedSignal` for proxies, it would be good to also support that in interfaces. This is likely a lot less work than it was for properties.

**Comment by mmasque** [comment]
Hi, I'm interested in working on this issue. I have been thinking a bit about an approach in `zbus_macros`. 
I have one design question. Suppose I have the following interface
```rust
#[dbus_interface(name = "org.zbus.MyGreeter1")]
impl Greeter {
    #[dbus_interface(property)]
    async fn foo(&self) -> u64 {
        0
    }
    #[dbus_interface(property)]
    async fn set_foo(&mut self, value: u64) -> zbus::fdo::Result<()> {
        self.foo = value;
        Ok(())
    }
```

- I think it makes sense to allow `emits_changed_signal` to be put in the `foo` header, and use that value to generate the signal function `foo_changed` when `emits_changed_signal` is `true` or `invalidate`. When set to `false` or `const`, the function could either a) still be generated and do nothing or b) not be generated at all.
- If we choose a) above, `set_foo` could always call `foo_changed`, and thus we wouldn't need to pass `emits_changed_signal` to the `set_foo` header. This isn't a very desirable approach, because in the default case we end up with a stub public function the user can call. 
- If we choose b) above, the implementation gets more complicated, because we need to only call the function `foo_changed` when it exists, and the macro code for `set_foo` needs a way to know whether the function exists or not. So there would be some refactoring to do to make sure that `foo` gets processed before `set_foo` and that we store whether the `foo_changed` function was generated.

Would like to hear your thoughts before I PR!

**Comment by zeenix** [comment]
> Would like to hear your thoughts before I PR!

Sure, sounds good to me. About your open question:

> * If we choose b) above, the implementation gets more complicated, because we need to only call the function `foo_changed` when it exists, and the macro code for `set_foo` needs a way to know whether the function exists or not. So there would be some refactoring to do to make sure that `foo` gets processed before `set_foo` and that we store whether the `foo_changed` function was generated.

That shouldn't be difficult and we have a lot of code in our macros that does similar things. It's actually what makes the macro code hard to modify for even small changes. :( If you refactor/clean up the code as part of your PR, that's even better. If you could refactor first in a separate commit before the real change (same PR), that would be great.

P.S. just to be sure we don't end up with any inconsistency, it's `invalidates` (`s` at the end).

**Comment by zeenix** [comment]
Was fixed a while back through #514.
